python/hsfs/hopsworks_udf.py

```diff
@@ -127,11 +145,21 @@ def __init__(
         return_types: Union[List[type], type, List[str], str],
         name: Optional[str] = None,
         transformation_features: Optional[List[TransformationFeature]] = None,
+        transformation_function_argument_names: Optional[
+            List[TransformationFeature]
+        ] = None,
+        dropped_argument_names: Optional[List[str]] = None,
+        dropped_feature_names: Optional[List[str]] = None,
+        feature_name_prefix: Optional[str] = None,
```

Comment:
this is needed for MDT only right?

Reply:
This is also required for on-demand features. 

So if the query used while creating a feature view uses prefix then all features from that feature group in the feature view would be prefixed with the name. So we need to add the prefix in the new output feature generated and also when trying to acess a specific feature for providing as input to the transformations function.

`query = fg1.select_all().join(fg2.select_all(), prefix="fg2_")`